### PR TITLE
CLI - Add `login show` and `logout`

### PR DESCRIPTION
**Pull request by bfops**
# Description of Changes

- Added the subcommand `login show` (to show the current login)
- Replaced `--clear-cache` with a separate `logout` subcommand (which imo is easier to understand)
- Made a small change to replace a `.conflicts_with` chain with an `ArgGroup`

# API and ABI breaking changes

Yes; this removes `--clear-cache`.

# Expected complexity level and risk

2

# Testing

- [x] `login clear` removed the logins from my config file
- [x] `login show` when not logged in gives an error message
- [x] `login show` when logged in shows my identity
- [x] `login show --token` when logged in shows my identity + token

### CODE REVIEW

**Review by cloutiertyler — APPROVED** [review]

File: crates/cli/src/subcommands/logout.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+use crate::Config;
+use clap::{ArgMatches, Command};
+
+pub fn cli() -> Command {
+    Command::new("logout")
+}
+
+pub async fn exec(mut config: Config, _args: &ArgMatches) -> Result<(), anyhow::Error> {
```
Comment (PuruVJ) [inline crates/cli/src/subcommands/logout.rs:8]: You also have to do `POST /auth/cli/logout` with same Authorizaton headers as sent on `/api/spacetimedb-token`, to delete this session from auth service also

**Review by PuruVJ — CHANGES_REQUESTED** [review]
Just one comment about logout

File: crates/cli/src/subcommands/logout.rs (hunk 1 above)
Comment (PuruVJ) [reply crates/cli/src/subcommands/logout.rs:8]: It also should take `auth-type` flag for that I assume

Comment (bfops) [reply crates/cli/src/subcommands/logout.rs:8]: @PuruVJ what's the `auth-type` flag? What would I pass?

Comment (PuruVJ) [reply crates/cli/src/subcommands/logout.rs]: Oh sorry I meant auth-host 😅

Comment (bfops) [reply crates/cli/src/subcommands/logout.rs]: ah okay :+1: in that case done

**Pull request merged**